docs/DeveloperGuide.md

```diff
@@ -4,10 +4,150 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+![architecture diagram](images/architecture_diagram.png)
+<br/> Fig 1 - Architecture Diagram
 
+The Architecture Diagram above shows the high-level design of the application. The **main components**
+consist of:
+- `MMM`: Initialises the components in the correct sequence and connects them with each other. Also holds the user's
+expenditures in memory.
+- `Ui`: The User Interface of the application and deals with interaction with the user.
+- `Parser`: Deals with making sense of user commands.
+- `Commands`: The collection of all executable commands.
+- `Storage`: Reads data from, and writes data to the hard disk.
+
+By abstracting out closely related code into classes, it allows`MMM`to deal at a higher level, without worrying
+about the lower level details. Higher cohesion is also achieved and coupling is minimized as each component is
+only coupled to the main class,`MMM`.
+
+The Sequence Diagram below shows an example of how the components interact with each other for the scenario 
+where the user issues the command`add shoes 100`.
+
+![sequence_diagram](images/sequence_diagram.png)
```

Comment:
Excellent diagram that shows a high-level overview of how the application works. Colouring to differentiate the different components are also good!

Reply:
For the sequence diagram, shouldn't the user go to :MMM with thecommand add shoes before the :Ui?
The sequence feels a bit off here

Reply:
I think they are trying to illustrate that the user enters the command through :UI, which is then sent to  :MMM to handle, which will then parse and do the rest of the sequence.